### PR TITLE
[MDS-5809] - updated error handling flow for incorrect IRT upload

### PR DESCRIPTION
**Pull request by matbusby-fw**
## Objective 

Updating to the import component caused the error reporting modal to not function properly.

This section of the code needs some love (and got a fair bit of it in this ticket, but didn't want to baloon scope too much).  Finding lots of odd choices here that are hard to follow and fix.

[MDS-5809](https://bcmines.atlassian.net/browse/MDS-5809)

image

### CODE REVIEW

File: services/minespace-web/src/components/Forms/projects/informationRequirementsTable/IRTFileImport.tsx [hunk 1]
```diff
@@ -77,21 +77,21 @@ export const IRTFileImport: FC<IRTFileImportProps> = ({
...
   ];
 
-  const handleCreateInformationRequirementsTable = (
+  const handleCreateInformationRequirementsTable = async (
     projectGuid: string,
     file: IFileInfo,
     documentGuid: string
   ) => {
-    dispatch(createInformationRequirementsTable(projectGuid, file, documentGuid));
+    return dispatch(createInformationRequirementsTable(projectGuid, file, documentGuid));
```
Comment (taraepp) [inline services/minespace-web/src/components/Forms/projects/informationRequirementsTable/IRTFileImport.tsx:85]: dispatch returns an action, doesn't it? It looks like it maybe wants to be await, but hard to tell in JS files.

**Review by henryoforeh-dev — APPROVED** [review]

File: services/minespace-web/src/components/Forms/projects/informationRequirementsTable/IRTFileImport.tsx (hunk 1 above)
Comment (matbusby-fw) [reply services/minespace-web/src/components/Forms/projects/informationRequirementsTable/IRTFileImport.tsx:85]: I thought so too, but I guess not.  ESLINT threw an error if I added an await there. (and it works without)

Comment (taraepp) [reply services/minespace-web/src/components/Forms/projects/informationRequirementsTable/IRTFileImport.tsx:85]: I guess the function consumer probably just wants something that _returns_? I find it odd but 🤷‍♀️ 
(I had to double-check, it does return an action)

**Review by taraepp — APPROVED** [review]

File: services/minespace-web/src/components/Forms/projects/informationRequirementsTable/IRTFileImport.tsx (hunk 1 above)
Comment (matbusby-fw) [reply services/minespace-web/src/components/Forms/projects/informationRequirementsTable/IRTFileImport.tsx:85]: Yeah, I did the same thing.  🤷🏻 

Honestly the setup here isn't great, but I think there are a lot of threads to pull here 😬
